extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -596,3 +636,69 @@ export async function promptSelectDatabase(connectionInfo: IConnectionInfo): Pro
 	}
 	return selectedDatabase;
 }
+
+export async function getConnectionURI(connectionInfo: IConnectionInfo): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+
+	let connectionURI: string = '';
+	try {
+		await vscode.window.withProgress(
+			{
+				location: vscode.ProgressLocation.Notification,
+				title: constants.connectionProgressTitle,
+				cancellable: false
+			}, async (_progress, _token) => {
+				// show progress bar while connecting to the users selected connection profile
+				connectionURI = await vscodeMssqlApi.connect(connectionInfo!);
+			}
+		);
+	} catch (e) {
+		// mssql connection error will be shown to the user
+		return undefined;
+	}
+
+	return connectionURI;
+}
+
+export async function promptSelectTable(connectionURI: string, bindingType: BindingType, selectedDatabase: string): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+	const userObjectName = bindingType === BindingType.input ? constants.enterObjectName : constants.enterObjectNameToUpsert;
+
+	// Create query to get list of tables from database selected
+	let tableQuery = tablesQuery(selectedDatabase);
+	const params = { ownerUri: connectionURI, queryString: tableQuery };
+	// send SimpleExecuteRequest query to STS to get list of schema and tables based on the connection profile of the user
+	let queryResult: azureFunctionsContracts.SimpleExecuteResult = await vscodeMssqlApi.sendRequest(azureFunctionsContracts.SimpleExecuteRequest.type, params);
+
+	// Get schema and table names from query result rows
+	const tableNames = queryResult.rows.map(r => r[0].displayValue);
+	// add manual entry option to table names list for user to choose from as well (with pencil icon)
+	let manuallyEnterObjectName = '$(pencil) ' + userObjectName;
+	tableNames.push(manuallyEnterObjectName);
+	// prompt user to select table from list of tables options
+	while (true) {
+		let selectedObject = await vscode.window.showQuickPick(tableNames, {
+			canPickMany: false,
+			title: constants.selectTable,
+			ignoreFocusOut: true
+		});
+
+		if (selectedObject === manuallyEnterObjectName) {
+			// user manually enters table or view to query or upsert into
+			selectedObject = await vscode.window.showInputBox({
+				prompt: bindingType === BindingType.input ? constants.sqlTableOrViewToQuery : constants.sqlTableToUpsert,
+				placeHolder: constants.placeHolderObject,
+				validateInput: input => input ? undefined : constants.nameMustNotBeEmpty,
+				ignoreFocusOut: true
+			});
+			if (!selectedObject) {
+				// user cancelled so we will show the tables prompt again
+				continue;
+			}
+		}
+
+		return selectedObject;
+	}
+}
+
+export function tablesQuery(selectedDatabase: string): string { return `SELECT QUOTENAME(CONCAT(table_schema,'.',table_name)) from ${selectedDatabase}.INFORMATION_SCHEMA.TABLES where TABLE_TYPE = 'BASE TABLE'`; }
```

Comment:
For the database (making a new comment here so it shows up in the files view)

If this query isn't being made with the selected database then that seems wrong to me and you should fix that. We should strive to open as few connections as possible.

(and separately from that - as with the table names you need to bracket-quote the server name if you end up keeping this. ANY time a user-entered value is inserted into a query like this then you need to correctly quote it or risk opening ourselves up to a SQL injection attack)

Reply:
From my understanding, the query utilizes the connection URI. 
So the workaround for this would either update the connectionInfo.database directly after the user selects one from the listDatabases call with the first connectionInfo call ie would have to duplicate this call https://github.com/microsoft/azuredatastudio/blob/90a22ff62ef53e66ab00676de8244c9ea19714a8/extensions/sql-bindings/src/common/azureFunctionsUtils.ts#L340. 
This is what I assume will get a new connectionURI to send to this tablesQuery. So from my understanding, this would result in the same number of connect calls. 
I will look into this further, but for now, think including the database is the best first step for us to get the appropriate result back since the fix would have to be in STS / vscode-mssql.

Reply:
For the last part in regards to bracket-quote. Am I supposed to also bracket quote the selectedDatabase? As if I do that then that would not return the query and give an error.

Reply:
Yes, you need to bracket quote that as well. (You can't use QUOTENAME though - that's not allowed. You'll need to do that one manually in the code here)

Reply:
@Charles-Gagnon I found I was not able to use concat('[', $(selectedDatabase),']') within the query and as a result had to make a separate variable to store this value. Just want to be sure this is what you mean in regards to quoting the value.